Source/dead.cpp

```diff
@@ -71,7 +71,7 @@ void InitDead()
 
 void AddDead(int dx, int dy, char dv, int ddir)
 {
-	dDead[dx][dy] = (dv & 0x1F) + (ddir << 5);
+	dDead[dx][dy] = (dv & 0x1F) + (ddir * 32);

```

Comment:
`dDead` is not an arithmetic type:

https://github.com/diasurgical/devilutionX/blob/177373d6746066ed3111d32923c7924d4c61380f/Source/gendung.cpp#L105-L111